Parsers/ASimAuthentication/Parsers/vimAuthenticationCiscoIse.yaml

```diff
@@ -0,0 +1,125 @@
+    | extend 
+            EventVendor = "Cisco"
+            , EventProduct = "ISE"
+            , EventProductVersion = "3.2"
+            , EventCount = int(1)
+            , EventSchema = "Authentication"
+            , EventSchemaVersion = "0.1.3"
+            , EventStartTime = coalesce(EventTime, TimeGenerated)
+            , EventEndTime = coalesce(EventTime, TimeGenerated)
+            , EventOriginalSeverity = extract(@".*(NOTICE|INFO|WARN|WARNING|ERROR|FATAL|DEBUG).*", 1, SyslogMessage)
+            , EventType = case(EventOriginalType in (logon_evts), "Logon", "Logoff")
+    | lookup EventName on EventOriginalType 
+    | extend FailureReason = tostring(extract(@"FailureReason=\d+\s(.*?)", 1, SyslogMessage))
+    | extend EventOriginalResultDetails = case(isnotempty(FailureReason), FailureReason, EventOriginalResultDetails)
+            , EventResult = case(EventOriginalType in (reslt_success_evts), "Success", "Failure")
+    | extend EventSeverity = case(EventResult == "Failure", "Low", "Informational")
+    | extend EventResultDetails = case(EventOriginalType in (freason_no_usr_passwd), "No such user or password",
+                                        EventOriginalType in (freason_no_usr), "No such user",
+                                        EventOriginalType in (freason_incorrect_passwd),"Incorrect password",
+                                        EventOriginalType in (freason_incorrect_key), "Incorrect key",
+                                        EventOriginalType in (freason_acc_expired), "Account expired",
+                                        EventOriginalType in (freason_passwd_expired), "Password expired",
+                                        EventOriginalType in (freason_user_locked), "User locked",
+                                        EventOriginalType in (freason_user_disabled), "User disabled",
+                                        EventOriginalType in (freason_policy_violation), "Logon violates policy",
+                                        EventOriginalType in (freason_session_expired), "Session expired",
+                                        EventOriginalType in (freason_other), "Other",
+                                        EventResult == "Success", "", "N/A")
+    //Dvc specific fields
+    | extend 
+          DvcIpAddr = extract(@"(d{1,3}\.\d{1.3}\.\d{1,3}\.\d{1,3})", 1, Computer)
+        , DvcHostname = coalesce(tostring(extract(@"NetworkDeviceName=(.*?),", 1, SyslogMessage)), Computer, HostName)
```

Comment:
Use parse-kv to extract the fields from Syslog message. Multiple extract statements would decrease the performance of the parser.

Reply:
changed to using parse-kv where it is applicable